File: Unmatched/Dtos/SidekickDto.cs
﻿namespace Unmatched.Dtos;

public class SidekickDto
{
    public virtual HeroDto Hero { get; set; }

    public Guid HeroId { get; set; }

    public int Hp { get; set; }

    public Guid Id { get; set; }

    public string Name { get; set; }
    
    public int Count { get; set; }
    
    public bool IsRanged { get; set; }
    
    public string MeleeRangeImageUrl => $"/{(IsRanged ? "Ranged" : "Melee")}.png";
}

File: Unmatched.UI.Angular/src/app/backend-api/models/sidekick-dto.ts
/* tslint:disable */
/* eslint-disable */
import { HeroDto } from './hero-dto';
export interface SidekickDto {
  count?: number;
  hero?: HeroDto;
  heroId?: string;
  hp?: number;
  id?: string;
  isRanged?: boolean;
  meleeRangeImageUrl?: null | string;
  name?: null | string;
}
